src/client/Microsoft.Identity.Client.Brokers/RuntimeBroker.cs

```diff
@@ -440,10 +430,44 @@ public bool IsBrokerInstalledAndInvokable(AuthorityType authorityType)
         /// <param name="account"></param>
         /// <returns></returns>
         /// <exception cref="NotImplementedException"></exception>
-        public Task RemoveAccountAsync(ApplicationConfiguration appConfig, IAccount account)
+        public async Task RemoveAccountAsync(ApplicationConfiguration appConfig, IAccount account)
         {
-            // TODO: needs to be implemeted
-            return Task.CompletedTask;
+            string correlationId = Guid.NewGuid().ToString();
+
+            _logger.Verbose("[WamBroker] Removing WAM Account.");
+
+            using (var core = new NativeInterop.Core())
+            using (var authParams = new NativeInterop.AuthParameters(appConfig.ClientId, account.Environment))
+            {
+                using (var wamAccount = await core.ReadAccountByIdAsync(
+                    account.HomeAccountId.ObjectId,
+                    correlationId).ConfigureAwait(false))
+                {
+                    if (wamAccount == null)
+                    {
+                        _logger.WarningPii(
+                            $"Could not find a WAM account for the selected user {account.Username}",
+                            "Could not find a WAM account for the selected user");
+
+                        throw new MsalServiceException("wam_failed", $"Could not find a WAM account for the selected user {account.Username}");
```

Comment:
If I remember correctly, MSAL does not throw exception if `pca.RemoveAsync(account)` results in a no-op, for example if there are no artefacts in the token cache related to the account. So I recommend that we do not throw exception here either, just log a warning.